enhancements/cluster-logging/cluster-logging-v2-apis.md

```diff
@@ -0,0 +1,429 @@
+* Automated migration path from v1 to v2
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal
+actually is. Describe clearly what will be changed, including all of
+the components that need to be modified and how they will be
+different. Include the reason for each choice in the design and
+implementation that is proposed here, and expand on reasons for not
+choosing alternatives in the Alternatives section at the end of the
+document.
+
+### Workflow Description
+
+The following workflow describes the first user story which is a superset of the others and allows deployment of a full logging stack to collect and forward logs to a Red Hat managed log store.
+
+**cluster administrator** is a human responsible for:
+
+* Managing and deploying day 2 operators
+* Managing and deploying an on-cluster LokiStack
+* Managing and deploying a cluster-wide log forwarder
+
+**obervability-operator** is an operator responsible for:
+
+* managing and deploying observability plugins (e.g log-view-plugin)
+
+**loki-operator** is an operator responsible for managing a loki stack
+
+**cluster-logging-operator** is an operator responsible for managing log collection and forwarding
+
+The cluster administrator does the following:
+
+1. Deploys the Red Hat **observability-operator**
+1. Deploys the Red Hat **loki-operator**
+1. Deploys an instance of **LokiStack** in the `openshift-logging` namespace
+1. Deploys the Red Hat **cluster-logging-operator**
+1. Creates a **ClusterLogForwarder** custom resource for the **LokiStack**
+
+The **loki-operator**:
+
+1. Deploys the logging-view-plugin for reading logs in the OpenShift console
+
+The **cluster-logging-operator**:
+
+1. Deploys the log collector to forward logs to log storage in the `openshift-logging` namespace
+
+### API Extensions
+
+CustomResourceDefinition:
+```yaml
+    apiVersion: "logging.openshift.io/v2"
+    kind: ClusterLogForwarder
+    metadata:
+      name: 
+    spec:
+      serviceAccount:
+        name: 
+        namespace:       #namespace of deployment and resources
+      inputs:
+      - name:
+        type:
+        application:
+          selector:  #labelselector
+          namespaces:
+            include: []
+            exclude: []
+          containers:
+            include: []
+            exclude: []
+            maxRecordsPerSecond: # map[containername] int 
+        infrastructure:
+          sources: []
+        audit:
+          sources: []
+        receiver:  
+      transforms:
+      - name:
+        type:              #kubeapiaudit, detectmultiline, parse, labels
+        kubeAPIAudit:
+        parse:
+      pipelines:
+       - inputRefs: []
+         outputRefs: []
+         transformRefs: []
+      outputs:
+      - name: 
+        type: 
+        url:
+        tls:
+        secret:
+        limitsPerForwarder:
+          maxRecordsPerSecond:
+        tuning:    # map[keyname] value. Should allow freeform buffer tuning?
```

Comment:
See https://github.com/openshift/enhancements/pull/1540 for all proposed tuning parameters

```suggestion
# Example
outputs:
  - name: detailed-tweaks
    tuning:      
  	  delivery: safe
	    compression: "zlib"  # Enable protocol-specific compression
	    reconnect:
	      maxDelay: 1s  # Wait max of 1 second between reconnect attempts
```
For v2 API move v1 outputs[].limits into tuning like this:
```
outputs:
  - tuning:      
	    rateLimit:
	      recordsPerSecond: #int - document per-forwarder/per-node multiplier
	      # policy: Drop|Block # may add in future, currently we only drop.
```